src/firewall/core/io/firewalld_conf.py

```diff
@@ -10,9 +10,53 @@
 import tempfile
 import shutil
 
+import firewall.errors
+import firewall.functions
 from firewall import config
 from firewall.core.logger import log
 
+
+def _validate_bool(value, default):
+    valid = True
+    try:
+        v = firewall.functions.str_to_bool(value, default=_validate_bool)
+    except ValueError:
+        valid = False
+        v = _validate_bool
+
+    if v is _validate_bool:
+        v = firewall.functions.str_to_bool(default)
+
+    return ("yes" if v else "no"), valid
+
+
+def _validate_enum(value, enum_values, default, is_lower_case=False):
+
+    for f in enum_values:
+        assert isinstance(f, str)
+        assert f == f.strip()
+        if is_lower_case:
+            assert f == f.lower()
+
+    assert default in enum_values
+
+    if value is None:
+        return default, True
+
+    if is_lower_case:
```

Comment:
Why do we need `is_lower_case`? Seems like we shouldn't care what case. Always normalize to lower and verify it in lower.

Reply:
it was for cases, where we would have case-sensitve enum values. I drop the (unused) parameter.